File: Data Structures & Algorithms/single-number-iii/submission-0.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:


        ans = 0

        for i in nums:
            ans^=i
        setbit = 1
        while (ans & setbit) == 0:
            setbit<<=1
        a = 0
        b = 0
        for i in nums:
            if i & setbit:
                a^=i
            else:
                b^=i
        return [a,b]

        

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {


        unordered_map<int,int> u;


        for(int i=0;i<nums.size() ;i++){
            if(u.find(target-nums[i])!=u.end()){
                return {u[target-nums[i]],i};
            }else{
                u[nums[i]] = i;
            }
        }


        return {-1,-1};
        
    }
};
